shared/views/RoomDirectoryView.js

```diff
@@ -169,8 +175,6 @@ class RoomDirectoryView extends TemplateView {
     t.mapSideEffect(
       (vm) => vm.homeserverSelection,
       (homeserverSelection, oldHomeserverSelection) => {
-        homeserverSelectElement.value = homeserverSelection;
```

Comment:
Removing this introduces a bug where the `<select>` isn't updated by the time we submit the form after selection just below.

We have a similar problem with the hash as well and we just set it directly before submitting. Which is the same concept as what this did.

The alternative is to do a `setTimeout(() => {...}, 0)` and hope that everything is up to date by the next loop.

Reply:
Fixed in https://github.com/matrix-org/matrix-public-archive/pull/105